### DISCUSSION TITLE
Free Monitor

### DISCUSSION THREAD

**Discussion by Ucoz3000** [RavenSystem/esp-homekit-devices#1717] [MEPLHAA Scripts]
10 and 11: ADC

Raw values come from Analog Pin (ADC). Those raw values are from 0 to 1023.

Inverted ADC means that raw value will be 1023 - ADC.

Example of a Free Monitor reading a MQ-135 sensor connected to ADC pin and setting the state of an Air Quality Service:

```{
  "a":[
    {
      "t" : 15,
    }, {
      "t" : 80,
      "h" : 0,
      "n" : 3,
      "y0" : [
        {
          "v" : 300,
          "0" : { "m" : [ [ -1, 1 ] ] }
        }, {
          "v" : 500,
          "0" : { "m" : [ [ -1, 2 ] ] }
        }, {
          "v" : 600,
          "0" : { "m" : [ [ -1, 3 ] ] }
        }, {
          "v" : 700,
          "0" : { "m" : [ [ -1, 4 ] ] }
        }, {
          "v" : 800,
          "0" : { "m" : [ [ -1, 5] ] }
        }
      ]
    }
  ]
}
```

Error: Parse error on line 4:
...     "t" : 15 ,    }, {      «т» : 80 
----------------------^
Expecting 'STRING', got '}'

**Comment by RavenSystem** [comment]
Fixed. Many thanks!

**Comment by Waniaok** [comment]
ADC is pin6/TOUT?
how to power the mq-135 5v or 3.3v?

**Reply by joddye2** [reply]
Only 5V, best to power it separately (not from ESP 5v pin (if u power it with onboard microusb - wemos/nodmcu)

**Reply by joddye2** [reply]
![38218D91-F8C5-4750-A883-0B3CB60858EE](https://user-images.githubusercontent.com/77693599/153087349-3f776f2c-e7fa-4109-912b-bc64e0488024.jpeg)
![Uploading DF38D176-83EE-4815-92E6-105AE9326B23.jpeg…]()
